app/javascript/controllers/concepts_json.js

```diff
@@ -0,0 +1,41 @@
+import {Controller} from "@hotwired/stimulus"
+
+export default class extends Controller {
+    static values = {
+        ontologyId: String,
+        conceptId: String,
+        apiKey: String
+      }
+
+    update(){
+        const baseClassUrl = `${this.ontologyIdValue}/classes/${this.conceptIdValue}`;
+        const apikey = `apikey=${this.apiKeyValue}`;
+        const tabsList = document.querySelectorAll('#concept_tabs_container .nav-item');
+        const links = {
+            details_tab: `${baseClassUrl}?${apikey}&display=all`,
```

Comment:
You can't use the IDs as identifiers as they can be changed, and it will break this without you knowing it. 
A solution is directly on each tab to specify its data-json-url="your URL" and if its empty don't show the json button.

Reply:
I don't understand what do you mean exactly here

Reply:
```
 const links = {
            details_tab: `${baseClassUrl}?${apikey}&display=all`,
            instances_tab: `${baseClassUrl}/instances?${apikey}`,
            notes_tab: `${baseClassUrl}/notes?${apikey}`,
            mappings_tab: `${baseClassUrl}/mappings?${apikey}`,
            visualization_tab: ''
        };
```
this is too specific, imagine I want to add a tab to the concept tabs, I will need to change here too, If we remember to.

in summary, make it generic.